src/js/angular/import/directives/import-resource-tree.directive.js

```diff
@@ -305,8 +305,37 @@ function importResourceTreeDirective($timeout, ImportContextService) {
             // Subscription handlers
             // =========================
 
-            const onResourcesUpdatedHandler = (resources) => {
-                $scope.resources = resources;
+            const onResourcesUpdatedHandler = (newResources = []) => {
+                const isUserImport = TABS.USER === ImportContextService.getActiveTabId();
+                if ($scope.resources.isEmpty()) {
+                    $scope.resources = ImportResourceTreeService.toImportResourceTree(_.cloneDeep(newResources), isUserImport);
```

Comment:
I don't get it why the cloneDeep was needed here? This onResourcesUpdatedHandler is invoked when resources are loaded from the backend and on the initial bootstrap where an empty array is passed. So, what's the reason to clone a backend response before converting it to tree model? Do you expect someone to work with the raw response instead with the tree?

Reply:
Because newResource is a parameter, we need to ensure the model is not changed. Actually, after these changes, we do not touch these models I will remove it.